lib/hl7/message.ex

```diff
@@ -529,6 +526,30 @@ defmodule HL7.Message do
     end
   end
 
+  def maybe_transcode(binary) do
+    case String.valid?(binary) do
+      true ->
+        binary
+
+      false ->
+        transcode(binary)
+    end
+  end
+
+  def transcode(binary) do
```

Comment:
I found this was the most performant method to accumulate. If that's so, then it may invalidate the need to perform a copy if that option is passed in.